src/PeriodEnd/PaymentsDue/SFA.DAS.ProviderPayments.Calc.PaymentsDue/Services/PaymentsDueCalculationService.cs

```diff
@@ -45,20 +40,21 @@ public List<RequiredPaymentEntity> CalculatePaymentsDue()
                     x.AccountId)
             ).ToDictionary(x => x.Key, x => x.ToList());
 
-            var groupedPastPayments = PastPayments.GroupBy(x => new MatchSetForPayments
-            (
-                x.StandardCode,
-                x.FrameworkCode,
-                x.ProgrammeType,
-                x.PathwayCode,
-                x.ApprenticeshipContractType,
-                x.TransactionType,
-                x.SfaContributionPercentage,
-                x.LearnAimRef,
-                x.FundingLineType,
-                x.DeliveryYear,
-                x.DeliveryMonth,
-                x.AccountId)).ToDictionary(x => x.Key, x => x.ToList());
+            var groupedPastPayments = pastPayments.GroupBy(x => new PaymentGroup
+                (
+                    x.StandardCode,
```

Comment:
all of these assignment lines can be moved to an implicit cast operator overload. and tested in isolation. the same goes for lines 29-40.